### PR TITLE
Sort mapillary's rbush result

### LINKED ISSUE #4297: Mapillary icons keep trying to overlay eachother
When dragging the map around at a higher zoom level, the mapillary icons seem to fight for who is on top.
screen

cc @kepta

### PR DESCRIPTION
**Pull request by kepta**
This sorts the data coming out of rbush.
(closes #4297)

### CODE REVIEW

**Comment by kepta** [comment]
I have added `console.time` for anyone to test the performance impact.
@JamesKingdom could share the url you are using in this .
https://user-images.githubusercontent.com/11473673/29823552-e65479f0-8cc6-11e7-90e2-d9beaf9f1711.gif

I am confused about the most performant sorting. 

**Case 1.** If I just sort the rbush search results, I can get a better performance. The problem would be the entire array wouldn't be sorted. Is this compromise stable enough to avoid the main problem in #4297 

**Case 2.** I can sort the entire array, with a little performance penalty. 

@bhousel I have tried to squeeze all the operations in one `reduce` hoping for a better performance over code legibility, thoughts?

File: modules/services/mapillary.js [hunk 1]
```diff
@@ -236,11 +236,32 @@ function searchLimited(psize, limit, projection, rtree) {
...
+    var x =  _.flatten(_.map(partitions, function(extent) {
         return rtree.search(extent.bbox())
             .slice(0, limit)
             .map(function(d) { return d.data; });
-    })));
+    }));
+    console.timeEnd('previous');
+
+    console.time('new');
+    var a = partitions.reduce(function(prev, curr) {
```
Comment (kepta) [inline modules/services/mapillary.js]: @bhousel I have tried to squeeze all the operations in one `reduce` hoping for a better performance over code legibility, thoughts?

Comment (bhousel) [reply modules/services/mapillary.js]: oh this looks great!  `reduce` is perfectly readable  👍

**Comment by bhousel** [comment]
This was merged 🚀 🤣

**Comment by bhousel** [comment]
I realized that we need to change some other things in order for this to really work.. see https://github.com/openstreetmap/iD/issues/4297#issuecomment-331924645

**Comment by bhousel** [comment]
BTW, deleting this branch as the code got merged as ddbc9df11

**Pull request closed without merging**